src/pages/EditReportFieldDropdownPage.tsx

```diff
@@ -46,33 +46,73 @@ function EditReportFieldDropdownPage({fieldName, onSubmit, fieldID, fieldValue,
     const [headerMessage, setHeaderMessage] = useState('');
 
     const sections = useMemo(() => {
-        const filteredRecentOptions = recentlyUsedOptions.filter((option) => option.toLowerCase().includes(searchValue.toLowerCase()));
-        const filteredRestOfOptions = fieldOptions.filter((option) => !filteredRecentOptions.includes(option) && option.toLowerCase().includes(searchValue.toLowerCase()));
-        setHeaderMessage(!filteredRecentOptions.length && !filteredRestOfOptions.length ? translate('common.noResultsFound') : '');
+        if (searchValue) {
+            const filteredOptions = fieldOptions.filter((option) => option.toLowerCase().includes(searchValue.toLowerCase()));
+            setHeaderMessage(!filteredOptions.length ? translate('common.noResultsFound') : '');
+            return [
+                {
+                    shouldShow: false,
+                    data: filteredOptions.map((option) => ({
+                        text: option,
+                        keyForList: option,
+                        searchText: option,
+                        tooltipText: option,
+                    })),
+                },
+            ];
+        }
 
-        return [
-            {
+        const selectedValue = fieldValue;
+
+        setHeaderMessage(!fieldOptions.length && !recentlyUsedOptions.length ? translate('common.noResultsFound') : '');
```

Comment:
To show the no result found text. I'm not sure if I answer your question 😆

Reply:
Isn't that case handled when search value is present and on L51 above?

Reply:
I can access a report field without a list, so I think we still need it for now, except we hide the report field if it doesn't have a list.
<img width="1126" alt="image" src="https://github.com/Expensify/App/assets/50919443/e2fb962a-c6c3-48e9-80eb-70f8f8f6f4a8">

Reply:
@MonilBhavsar  Can you confirm the expected behavior when the report field is empty? We should still show the report field, allow user to click on it and it will show no result found (like current code) or we should remove or disable report field when it's empty?

Reply:
If there are no options, then i think we should disable the menu item and prevent clicking on it and rendering the page.
I think @thienlnam @allroundexperts would know better as they worked on this project.

Reply:
If this is a report field, and every dropdown is disabled we should actually not show the report field. This is consistent with what we do in OldDot

Reply:
So, are we going to handle it here? I think it's out of scope though.

Reply:
I think we can. We would be able to remove the line 67, if we're not rendering the selection page when there are no fields

Reply:
> We would be able to remove the line 67, if we're not rendering the selection page when there are no fields

But we currently show the report field even when it's empty. Do we still want to remove the line 67?

Reply:
Oh wait.. we still need that line because we still have the case that there's no result while searching.

Reply:
@hungvu193 that's handled in line 51

Reply:
ah ok, I see 🥲

Reply:
So, what do we do here? I don't think it's completely out of scope, as line 67 is of no use, if we don't open the options menu when there are no options. Or we can put this on HOLD until we have functionality to disable the menu when there are no options

Reply:
So if we decide to do it here, there will be few cases that we need to handle:
- when user tried to access the report field from deeplink
- when you're at the report field page and user delete all the report fields from old dot.

Reply:
Hiding the report field when it's empty is still out of scope for me.

If we still want to do it, we need to do 2 changes:

1. Filter out empty dropdown
https://github.com/Expensify/App/blob/0dc2151378a3fdb4328cb3e0f1e5eb2dbc8e1e7c/src/libs/ReportUtils.ts#L2096-L2110
```
return mergedFieldIds.map((id) => report?.reportFields?.[id] ?? policyReportFields.find(({fieldID}) => fieldID === id)).filter((reportField) => {
    if (reportField?.type === 'dropdown') return reportField?.values.length > 0;
    return true;
}) as PolicyReportField[];
```

2. Show the not found page for empty dropdown
https://github.com/Expensify/App/blob/0dc2151378a3fdb4328cb3e0f1e5eb2dbc8e1e7c/src/pages/EditReportFieldPage.tsx#L49-L63
```
if (!reportField || !report || isDisabled || (reportField.type === 'dropdown' && reportField.values.length === 0)) {
```

And I think bounty increase would be fair.

Reply:
hey @MonilBhavsar, how do you think about @bernhardoj opinion above?

Reply:
Ah okay, fair amount of changes to be in the scope. Fine keeping it for next PR.

Reply:
Thanks for confirming, @bernhardoj Can you update the PR so we can move it forward? 🙇

Reply:
Oh wait, @MonilBhavsar do you mean to include it in this PR or not?

>  Fine keeping it for next PR.

I read this and think that we will just skip it in this PR.

Reply:
Yes, sorry if it was unclear. Let's not include those changes in this PR.

Reply:
So let remove the line 67 and other changes will be in the next PR right?

Reply:
Cool!

Reply:
Done